packages/compass-explain-plan/src/components/explain-tree/explain-tree.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useRef, useEffect } from 'react';
+import { map } from 'lodash';
+import d3 from 'd3';
+import { ExplainStage } from '../explain-stage';
+import {
+  css,
+  palette,
+  spacing,
+  useDarkMode,
+} from '@mongodb-js/compass-components';
+
+import STAGE_CARD_PROPERTIES from '../../constants/stage-card-properties';
+
+interface ExplainTreeProps {
+  nodes: any[];
+  links: any[];
+  width: number;
+  height: number;
+}
+
+const explainTreeStyles = css({
+  position: 'relative',
+  zIndex: 0,
+  margin: `${spacing[5]}px auto`,
+});
+
+const ExplainTree: React.FunctionComponent<ExplainTreeProps> = ({
+  nodes,
+  links,
+  width,
+  height,
+}) => {
+  const svgRef = useRef<SVGSVGElement>(null);
+  const darkMode = useDarkMode();
+
+  useEffect(() => {
+    if (!svgRef.current || !links || !nodes) return;
+    const svgElement = svgRef.current;
+
+    // Right angle links between nodes
+    const elbow = (d: {
+      source: { x: number; y: number; x_size: number };
+      target: { y: number; x: number; key: string };
+    }): string => {
+      return `M${d.source.x + d.source.x_size / 2},${d.source.y}
+        V${d.target.y - STAGE_CARD_PROPERTIES.VERTICAL_PADDING / 2}
+        H${d.target.x + STAGE_CARD_PROPERTIES.DEFAULT_CARD_WIDTH / 2}
+        V${d.target.y}`;
+    };
```

Comment:
If you are already building the path yourself, why do we need d3 here at all? Why not just render it with React?

Reply:
that much I don't know is a copy-paste of the code that was there, I'll give this file a second pass and address it in a different PR, the code to layout the tree is spread here and there, I'd love to take a bit more time and make a dedicate change

Reply:
Gotcha, makes sense!